holidays/ical.py

```diff
@@ -0,0 +1,154 @@
+
+        Args:
+        - holidays_object: Holidays object containing holiday data.
+        - ical_timestamp: Optional override for the iCal "DTSTAMP" parameter.
+            If not provided, current datetime (UTC timezone) will be used instead.
+        """
+        self.holidays = holidays_object
+        self.ical_timestamp = ical_timestamp or datetime.now(timezone.utc).strftime(
+            "%Y%m%dT%H%M%SZ"
+        )
+        self.holidays_version = __version__
+        self.language = self._validate_language(
+            getattr(self.holidays, "language", None)
+            or getattr(self.holidays, "default_language", None)
+            or "EN"
+        )
+
+    def _validate_language(self, language):
+        """
+        Validates the language code. Ensures it's a two-letter ISO 639-1 code
+        as iCal only supports that at the moment.
+
+        Args:
+        - language: The language code to validate.
+
+        Returns:
+        - Valid two-letter language code in uppercase.
+        """
+        language = language.upper().split("_")[0]
+
+        if len(language) != 2:
+            raise ValueError(
+                f"Invalid language code: {language}. "
+                "Only two-letter ISO 639-1 codes are supported by iCal. "
+                "Refer to the 'ISO 639-1 Code' column at "
+                "https://www.loc.gov/standards/iso639-2/php/code_list.php for valid codes."
+            )
+        return language
+
+    def _fold_line(self, line):
+        """
+        Fold long lines according to RFC 5545.
+
+        Lines of text SHOULD NOT be longer than 75 octets.
+        """
+        if len(line.encode("utf-8")) <= 75:
+            return [line]
+
+        result = [line]
+        while len(result[-1].encode("utf-8")) > 75:
+            current = result.pop()
+            pos = 74
+            while len(current[:pos].encode("utf-8")) > 74:
+                pos -= 1
+            result.append(current[:pos])
+            result.append(f" {current[pos:]}")
+
+        return result
+
+    def _generate_event(self, date, holiday_name, holiday_length: int = 1):
+        """
+        Generate a single holiday event.
+
+        Args:
+        - date: Holiday date.
+        - holiday_name: Holiday name.
+        - holiday_length: Holiday length in days, default to 1.
+
+        Returns:
+        - list[str]: List of iCalender format event lines.
+        """
+        formatted_date = date.strftime("%Y%m%d")
+        event_uid = f"{uuid.uuid4()}@{self.holidays_version}.holidays.local"
+
+        lines = [
+            "BEGIN:VEVENT",
+            f"DTSTAMP:{self.ical_timestamp}",
+            f"UID:{event_uid}",
+            f"SUMMARY:{holiday_name}",
+            f"DTSTART;VALUE=DATE:{formatted_date}",
+            f"DURATION:P{holiday_length}D",
+            "END:VEVENT",
+        ]
+
+        folded_lines = []
+        for line in lines:
+            folded_lines.extend(self._fold_line(line))
```

Comment:
It's better to process `_fold_line()` only summary string, because the rest are clearly shorter.